File: yardiel.py
import csv
import numpy as np
import tensorflow as tf
import random as rn
import os
os.environ['PYTHONHASHSEED'] = '0'
np.random.seed(41)
rn.seed(41)
tf.random.set_seed(41)

from tensorflow import keras
from keras import optimizers
from keras import models
from keras import layers
from keras import losses
from keras import metrics
from keras.models import Sequential
from keras.layers import Dense
from keras.models import load_model

# se carga la libreria de tensorflow
import tensorflow.compat.v1 as tf

# se carga el aditamento para seleccionar el gpu
import os

os.environ ['CUDA_VISIBLE_DEVICES'] = '0'

#escoger porcentage de memorya en gpu para ser usada
#config.gpu_options.per_process_gpu_memory_fraction=0.9
config = tf.ConfigProto()
config.gpu_options.allow_growth = True

# despues de la configuracion validar, se inicia la sesion
sess = tf.Session (config = config)
import time
start_time = time.time()




archivo = "/Users/natorro/Desktop/programming/yardiel_projects/hawc-events/datasets/hawc_data/hawc_crudos200k.csv"

archivo = open(archivo)
csvreader = csv.reader(archivo)

vec=[]

next(csvreader)
for i in range(160000):
    vec.append(next(csvreader))


print(len(vec[8]))


clase = []
for i in vec:
    clase.append(i[len(i)-1])
    

clase[0]

salida = []
for i in range(160000):
    salida.append([])
    salida[i].append(clase[i])

print(salida)
print(clase)
elementos = []

count = 0
for i in vec:
    i.pop(0)
    i.pop(300)
    count += 1
    elementos.append(i)

print(elementos[0])
len(elementos[0])




entrada = np.array(elementos, "float32")
salidas = np.array(salida, "float32")

model = models.Sequential()
model.add(layers.Dense(300, activation = 'sigmoid', input_shape = (300, )))
model.add(layers.Dense(300, activation = 'relu'))
model.add(layers.Dense(300, activation = 'sigmoid'))
model.add(layers.Dense(100, activation = 'sigmoid'))
model.add(layers.Dense(1, activation = 'sigmoid'))

model.compile(optimizer = optimizers.RMSprop(learning_rate = 0.01), 
              loss = losses.binary_crossentropy, 
              metrics = [metrics.binary_accuracy])
model.fit(entrada, salidas, epochs = 200)



model.save('modelodia1.h5')
model1=load_model('modelodia1.h5')

print("\n \n \n Time to finish: --- %s seconds ---" % (time.time() - start_time))

File: yardiel.r
#install.packages("keras", dependencies = TRUE)
library(keras)

rawData <- read.csv("/Users/natorro/Desktop/programming/yardiel_projects/hawc-events/datasets/hawc_data/hawc_crudos200k.csv")

rawData<- subset(rawData, select = -c(evento))
indexes <- sample(1:200000, 40000)
randomSampleData <- rawData[indexes, ]
leftData <- rawData[-indexes, ]

training_events_df <- subset(randomSampleData, select = -c(class))
training_events_df <- as.matrix(training_events_df)
dimnames(training_events_df) <- NULL 

training_class_df <- subset(randomSampleData, select = c(class))
training_class_df <- as.matrix(training_class_df)
dimnames(training_class_df) <- NULL

test_events_df <- subset(leftData, select = -c(class))
test_events_df <- as.matrix(test_events_df)
dimnames(test_events_df) <- NULL

test_class_df <- subset(leftData, select = c(class))
test_class_df <- as.matrix(test_class_df)
dimnames(test_class_df) <- NULL


# DO I NEED TO RESHAPE HERE?
train_evs <- array_reshape(training_events_df, c(40000, 300))
train_class <- array_reshape(training_class_df, c(40000, 1))

rm("rawData", "indexes", "randomSampleData", "leftData")


network <- keras_model_sequential() %>%
      layer_dense(units = 300, activation = "sigmoid", input_shape = c(300)) %>% 
      layer_dense(units = 300, activation = "sigmoid") %>% 
      layer_dense(units = 300, activation = "sigmoid") %>% 
      layer_dense(units = 100, activation = "sigmoid") %>% 
      layer_dense(units = 1, activation = "sigmoid")

network %>% compile(
      optimizer = optimizer_rmsprop(learning_rate = 0.001),
      loss = "binary_crossentropy", 
      metrics = c("binary_accuracy")
)

network %>% fit(train_evs, train_class, epochs = 200)

save_model_hdf5(network, 
                filepath = "/Users/natorro/Desktop/programming/yardiel_projects/hawc-events/fivelayers2.hf5")
      